Outdated embedded redis dependency causes test failures as redis does not start

**Problem description**
Both RedisQosProfileServiceTest and SessionsControllerIntegrationTest fails as embedded redis does not startup

**Expected action**

1. Run mvn clean test
2. In the logs you will see
[ERROR] Errors:
[ERROR]   SessionsControllerIntegrationTest.setUp:158 » Runtime Can't start redis server. Check logs for details. Redis process log:
[ERROR]   RedisQosProfileServiceTest.setUp:61 » Runtime Can't start redis server. Check logs for details. Redis process log:
[INFO]
[ERROR] Tests run: 74, Failures: 0, Errors: 2, Skipped: 0
3. Caused by [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.119 s <<< FAILURE! -- in com.camara.qod.service.RedisQosProfileServiceTest
[ERROR] com.camara.qod.service.RedisQosProfileServiceTest -- Time elapsed: 0.119 s <<< ERROR!
java.lang.RuntimeException: Can't start redis server. Check logs for details. Redis process log:
	at redis.embedded.AbstractRedisInstance.awaitRedisServerReady(AbstractRedisInstance.java:70)
	at redis.embedded.AbstractRedisInstance.start(AbstractRedisInstance.java:42)
	at redis.embedded.RedisServer.start(RedisServer.java:9)
	at com.camara.qod.service.RedisQosProfileServiceTest.setUp(RedisQosProfileServiceTest.java:61)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)

**Additional context**
This is happening on MacOS 14.5 and openjdk 17.0.2 2022-01-18

--- a/core/pom.xml
+++ b/core/pom.xml
@@ -112,9 +112,9 @@
       <version>${shedlock-spring.version}</version>
     </dependency>
     <dependency>
-      <groupId>it.ozimov</groupId>
-      <artifactId>embedded-redis</artifactId> <!-- Embedded Redis for Unit Tests-->
-      <version>0.7.3</version>
+      <groupId>com.github.codemonstur</groupId>
+      <artifactId>embedded-redis</artifactId>
+      <version>1.4.3</version>
       <scope>test</scope>
     </dependency>
     <dependency>

--- a/core/src/test/java/com/camara/qod/controller/SessionsControllerIntegrationTest.java
+++ b/core/src/test/java/com/camara/qod/controller/SessionsControllerIntegrationTest.java
@@ -79,6 +79,7 @@ import com.camara.qod.service.ExpiredSessionMonitor;
 import com.camara.qod.service.StorageService;
 import com.fasterxml.jackson.core.JsonProcessingException;
 import feign.FeignException;
+import java.io.IOException;
 import java.net.URI;
 import java.nio.charset.StandardCharsets;
 import java.time.Instant;
@@ -150,7 +151,7 @@ class SessionsControllerIntegrationTest {
   static final int MAX_VALUE_SECONDS_PER_DAY = 86399;
 
   @BeforeAll
-  public static void setUp() {
+  public static void setUp() throws IOException {
     redisServer = new RedisServer(6370); // RedisServer.builder().port(6370).setting("maxheap 128M").build();
     if (redisServer.isActive()) {
       redisServer.stop();
@@ -159,7 +160,7 @@ class SessionsControllerIntegrationTest {
   }
 
   @AfterAll
-  public static void tearDown() {
+  public static void tearDown() throws IOException {
     redisServer.stop();
   }
 

--- a/core/src/test/java/com/camara/qod/service/RedisQosProfileServiceTest.java
+++ b/core/src/test/java/com/camara/qod/service/RedisQosProfileServiceTest.java
@@ -29,6 +29,7 @@ import static org.junit.jupiter.api.Assertions.assertThrows;
 import com.camara.qod.api.model.QosProfile;
 import com.camara.qod.api.model.QosProfileStatusEnum;
 import com.camara.qod.exception.QodApiException;
+import java.io.IOException;
 import java.util.List;
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.BeforeAll;
@@ -53,7 +54,7 @@ class RedisQosProfileServiceTest {
   private static RedisServer redisServer = null;
 
   @BeforeAll
-  public static void setUp() {
+  public static void setUp() throws IOException {
     redisServer = new RedisServer(6370);
     if (redisServer.isActive()) {
       redisServer.stop();
@@ -62,7 +63,7 @@ class RedisQosProfileServiceTest {
   }
 
   @AfterAll
-  public static void tearDown() {
+  public static void tearDown() throws IOException {
     redisServer.stop();
   }
 

